File: app/src/main/java/ua/com/todd/smsforwadinga/smsforwading/managers/PreferenceManager.java
package ua.com.todd.smsforwadinga.smsforwading.managers;

import ua.com.todd.baseapp.managers.BasePreferenceManager;
import ua.com.todd.smsforwadinga.smsforwading.MyApplication;

public class PreferenceManager extends BasePreferenceManager<MyApplication> {

    enum Keys implements IKey {
        EMAIL, PASS, PORT, HOST
    }

    public PreferenceManager(MyApplication app) {
        super(app);
    }

    public void storeEmail(String str) {
        storeValue(Keys.EMAIL, str);
    }

    public String restoreEmail() {
        return restoreString(Keys.EMAIL);
    }

    public void storePassword(String str) {
        storeValue(Keys.PASS, str);
    }

    public String restorePassword() {
        return restoreString(Keys.PASS);
    }

    public void storePort(int value) {
        storeValue(Keys.PORT, value);
    }

    public Integer restorePort() {
        return restoreInt(Keys.PORT);
    }

    public void storeHost(String str) {
        storeValue(Keys.HOST, str);
    }

    public String restoreHost() {
        return restoreString(Keys.HOST);
    }
}

File: app/src/main/kotlin/ua/com/todd/smsforwading/managers/PreferenceManager.kt
package ua.com.todd.smsforwading.managers

import ua.com.todd.baseapp.managers.BasePreferenceManager
import ua.com.todd.smsforwading.MyApplication

public class PreferenceManager(app: MyApplication) : BasePreferenceManager<MyApplication>(app) {

    enum class Keys : BasePreferenceManager.IKey {
        EMAIL
        PASS
        PORT
        HOST
        IS_ENABLED
    }

    public fun storeEnabled(b : Boolean):Unit = storeValue(Keys.IS_ENABLED, b)

    public fun isEnabled(): Boolean = restoreBoolean(Keys.IS_ENABLED)

    public fun storeEmail(str: String): Unit = storeValue(Keys.EMAIL, str)

    public fun restoreEmail(): String = restoreString(Keys.EMAIL)

    public fun storePassword(str: String): Unit = storeValue(Keys.PASS, str)

    public fun restorePassword(): String = restoreString(Keys.PASS)

    public fun storePort(value: Int): Unit = storeValue(Keys.PORT, value)

    public fun restorePort(): Int = restoreInt(Keys.PORT)

    public fun storeHost(str: String): Unit = storeValue(Keys.HOST, str)

    public fun restoreHost(): String = restoreString(Keys.HOST)
}
